bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/cache/TTSLRUCacheImpl.java

```diff
@@ -0,0 +1,371 @@
+            logger.error("Cannot compute free disk space for the TTS cache. Reason: {}", e.getMessage());
+            return 0;
+        }
+    }
+
+    @Override
+    public AudioStream getOrSynthetize(TTSService tts, String text, Voice voice, AudioFormat requestedFormat)
+            throws TTSException {
+        checkFolder();
+
+        // initialize the supplier stream from the TTS service :
+        AudioStreamSupplier ttsSynthetizerSupplier = new AudioStreamSupplier(tts, text, voice, requestedFormat);
+        this.lock.lock(); // (a get operation also need the lock as it will update the head of the cache)
+        try {
+            String key = tts.getClass().getSimpleName() + "_" + tts.getCacheKey(text, voice, requestedFormat);
+            // try to get from cache
+            TTSResult ttsResult = get(key);
+            if (ttsResult == null || !ttsResult.getText().equals(text)) { // it's a cache miss or a false positive, we
+                                                                          // must (re)create it
+                logger.debug("Cache miss {}", key);
+                ttsResult = new TTSResult(cacheFolder, key, ttsSynthetizerSupplier);
+                put(key, ttsResult);
+            } else {
+                logger.debug("Cache hit {}", key);
+            }
+            return ttsResult.getAudioStreamClient(ttsSynthetizerSupplier);
+        } finally {
+            this.lock.unlock();
+        }
+    }
+
+    /**
+     * Load all {@link TTSResult} cached to the disk.
+     */
+    private void loadAll() {
+        ttsResultMap.clear();
+
+        try (Stream<Path> stream = Files.list(cacheFolder.toPath())) {
+            // Load the TTSResult from cache directory, order by date
+            List<@Nullable TTSResult> ttsResultOrderedList = stream
+                    .filter(path -> path.getFileName().toString().endsWith(INFO_EXT)).map(Path::toFile)
+                    .sorted((file1, file2) -> Long.valueOf(file1.lastModified() - file2.lastModified()).intValue())
+                    .map(this::buildFromFile).filter(Objects::nonNull).collect(Collectors.toList());
+
+            // Now create links between ordered entries in the list :
+            TTSResult previous = null;
+            for (TTSResult current : ttsResultOrderedList) {
+                if (current != null) {
+                    if (head == null) {
+                        head = current;
+                    }
+                    if (tail == null) {
+                        tail = current;
+                    }
+
+                    current.setPrevious(previous);
+                    if (previous != null) {
+                        previous.setNext(current);
+                    }
+                    previous = current;
+                }
+            }
+
+            ttsResultMap = ttsResultOrderedList.stream()
+                    .collect(Collectors.toMap(TTSResult::getKey, Function.identity()));
```

Comment:
```suggestion
                    .collect(Collectors.toMap(TTSResult::getKey, v -> v));
```

Reply:
Strangely enough, if I do this, I have a compilation error about a Null type mismatch ?